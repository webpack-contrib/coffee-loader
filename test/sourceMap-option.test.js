import {
  compile,
  execute,
  getCompiler,
  getErrors,
  getWarnings,
  readAsset,
} from './helpers';

describe('"sourceMap" option', () => {
  it('should not generate source maps with "Boolean" value equal "false"', async () => {
    const compiler = getCompiler('simple.js', {
      sourceMap: false,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(getErrors(stats)).toMatchSnapshot('errors');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
  });

  it('should generate source maps with "Boolean" value equal "true"', async () => {
    const compiler = getCompiler('simple.js', {
      sourceMap: true,
    });
    const stats = await compile(compiler);

    expect(
      execute(readAsset('main.bundle.js', compiler, stats))
    ).toMatchSnapshot('result');
    expect(getErrors(stats)).toMatchSnapshot('errors');
    expect(getWarnings(stats)).toMatchSnapshot('warnings');
  });
});
