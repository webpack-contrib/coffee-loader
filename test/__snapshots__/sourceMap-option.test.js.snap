// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"sourceMap" option should generate source maps with "Boolean" value equal "true": errors 1`] = `Array []`;

exports[`"sourceMap" option should generate source maps with "Boolean" value equal "true": result 1`] = `
Object {
  "cubes": Array [
    1,
    8,
    27,
    64,
    125,
  ],
  "list": Array [
    1,
    2,
    3,
    4,
    5,
  ],
  "local": Object {
    "default": [Function],
  },
  "math": Object {
    "cube": [Function],
    "root": [Function],
    "square": [Function],
  },
  "number": -42,
  "opposite": true,
  "race": [Function],
  "square": [Function],
}
`;

exports[`"sourceMap" option should generate source maps with "Boolean" value equal "true": warnings 1`] = `Array []`;

exports[`"sourceMap" option should not generate source maps with "Boolean" value equal "false": errors 1`] = `Array []`;

exports[`"sourceMap" option should not generate source maps with "Boolean" value equal "false": result 1`] = `
Object {
  "cubes": Array [
    1,
    8,
    27,
    64,
    125,
  ],
  "list": Array [
    1,
    2,
    3,
    4,
    5,
  ],
  "local": Object {
    "default": [Function],
  },
  "math": Object {
    "cube": [Function],
    "root": [Function],
    "square": [Function],
  },
  "number": -42,
  "opposite": true,
  "race": [Function],
  "square": [Function],
}
`;

exports[`"sourceMap" option should not generate source maps with "Boolean" value equal "false": warnings 1`] = `Array []`;
